### ISSUE TITLE
Product a 'Dart Platform SDK' on the Dart bots

### ISSUE THREAD

**Issue by mit-mit** [dart-lang/sdk#29558]
## Motivation

Currently the Dart SDK contains several libraries and tools that are web specific. For products that embed the Dart SDK but which are not web oriented -- e.g. http://flutter.io -- these take up space, yet provide no value. It would be good to produce a more focused Platform SDK for these use cases.

## Contents

The contents of the platform SDK should be the contents of the current Dart SDK with the following _not included_:

* /bin/
  - `dart2js`
  - `snapshots/dart2js.dart.snapshot`
  - `dartdevc`
  - `snapshots/dartdevc.dart.snapshot`
* /lib/
  - `_blink`
  - `_chrome`
  - `html`
  - `indexed_db`
  - `js`
  - `js_util
  - `mirrors`
  - `svg`
  - `web_audio`
  - `web_gl`
  - `web_sql`

## Production

This platform SDK should be produced on the bots just like the current Dart SDK. It should also be uploaded to cloud storage like the current SDK.

## Naming

The name of the produced .zip file should be `dart-platform-sdk-<platform>-<architecture>-release.zip`

The file should be uploaded to the storage bucket `https://storage.googleapis.com/dart-archive/channels/<stable/dev>/release/<release>/platform-sdk/`.

**Comment by mit-mit** [comment]
@kevmoo can you sanity check exclude lists above, please?

**Comment by rmacnak-google** [comment]
Note if you include the standalone VM (`bin/dart`) in this SDK variant, you should still keep `lib/mirrors`.

**Comment by mit-mit** [comment]
Ah, good catch, updated!

**Comment by whesse** [comment]
@zanderso @rmacnak-google 
The work to move the creation of the SDK to GN means that this will be created as a GN target, right?

**Comment by zanderso** [comment]
It's a bit simpler to choose the smaller SDK with GN argument rather than creating a new target for it, so that's what I'd like to do. I can wire up the GN argument to an environment variable in gn.py so that it's easier to set up on the bots.

Is it also possible to exclude the analyzer, analysis_server, front_end, and kernel sources in the small SDK?

https://codereview.chromium.org/2848943003/

**Comment by mit-mit** [comment]
> Is it also possible to exclude the analyzer, analysis_server, front_end, and kernel sources in the small SDK?

I think Flutter needs those; @devoncarew?

**Comment by devoncarew** [comment]
Yup, at this point those are specifically there for use by Flutter. They use the sources from the sdk in preference to hosted - pub.dartlang.org - versions (@pq to confirm).

**Comment by zanderso** [comment]
Alright. I've updated the CL to make sure the sources are in there. The CL is ready for review, but I'll be OOO next week. I'll be able to address comments and land when I get back.

**Comment by pq** [comment]
> They use the sources from the sdk in preference to hosted - pub.dartlang.org - versions (@pq to confirm).

👍 Spot on.

**Comment by zanderso** [comment]
After the above commit, when `DART_MAKE_PLATFORM_SDK` is in the environment on a `gclient sync`, or when `--platform-sdk` is passed to `tools/gn.py`, only the subset of the SDK described above will be built.

**Comment by jacob314** [comment]
Any objections to also setting 
dart_platform_sdk=true for all arm and arm64 builds? I don't think we need to be building dart2js and web libraries arm and arm64?

**Comment by jacob314** [comment]
https://codereview.chromium.org/2900833002

**Comment by whesse** [comment]
We release the arm and arm64 sdks on the Dartlang.org page.  I'm not sure
why people downloading the sdk wouldn't need dart2js.  I could be convinced
though.  So only VM and analyzer and formatter would be built and tested?

**Comment by zanderso** [comment]
I think that sounds reasonable. If we hear that someone wants to do self-hosted web development on a raspberry pi or similar, we can revisit.

**Comment by mit-mit** [comment]
@zanderso didn't we complete this?

**Comment by zanderso** [comment]
VM and build work is done. This should be assigned to infra to produce and upload the artifact on the bots.

**Comment by athomas** [comment]
Not a priority until @mit-mit says it is.

**Comment by zanderso** [comment]
Flutter users currently download the smaller platform SDK. It is built by the Flutter engine builders. I agree that it is probably lower priority now for our bots to do it. However, it would still be nice to have a smaller download of the prebuilt SDK for Fuchsia since they don't vend a Dart SDK from anywhere for their own consumption.

@mit-mit Are there any thoughts about when this issue might get addressed?

**Comment by sedlund** [comment]
I downloaded the dart-sdk zip file from the archive page for my ARM Chromebook (running crouton debian) as there is no deb being built.  In doing the angular quickstart, I found that dartdevc was not included in the zip file which lead me here.

**Comment by devoncarew** [comment]
To follow up on this, I don't believe that flutter needs the analyzer, analysis_server, front_end, ... sources in the sdk any longer.
